data/transactions/logic/eval.go

```diff
@@ -1335,7 +1345,12 @@ func opGtxn(cx *evalContext) {
 		return
 	}
 	tx := &cx.TxnGroup[gtxid].Txn
-	field := uint64(cx.program[cx.pc+2])
+	field := TxnField(uint64(cx.program[cx.pc+2]))
+	fs, ok := txnFieldSpecByField[field]
+	if !ok || fs.version > cx.version {
+		cx.err = fmt.Errorf("invalid txn field %d", field)
+		return
+	}
 	var sv stackValue
```

Comment:
Should `txn ApplicationArgs` and `gtxn 0 ApplicationArgs` fail?

Reply:
Nice catch, fixed.